freedomofpress/securedrop-client
# Add action to (re-)generate and export conversation transcript

## Description

The shortest path to providing an option to export a conversation transcript to journalists.

## Context

This is a direct continuation of #1621, the same context applies.

## Implementation

- Takes the existing _export dialog_ as a reference
- Follows the existing patterns for testing it

## Test plan

1. Start the app
2. Select a conversation
3. For good measure download a file or two
4. Trigger "Export Conversation Transcript" in the _conversation menu_ (a.k.a source menu, kebab menu...)
- [ ] Confirm that the dialog says it will export `conversation.txt`
- [ ] Confirm the file is exported as expected
- [ ] Confirm that error messages are shown in the same ways as in the existing export dialog

(I can provide guidance for a lightweight tests like I did for #1621, but since exporting doesn't depend on hard to get hardware, I think we should eventually test the feature using an actual USB drive. **Edit**: here is the [patch](https://github.com/freedomofpress/securedrop-client/commit/c3f6f80394b5567163747a094227469e88791a9a).)
